src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -175,38 +211,88 @@ public void FromStream(Deserializer deserializer)
 
             sessionStartTimeQPC = _syncTimeQPC;
 
-            if (deserializer.VersionBeingRead >= 3)
+            if (3 <= deserializer.VersionBeingRead)
             {
                 deserializer.Read(out pointerSize);
                 deserializer.Read(out _processId);
                 deserializer.Read(out numberOfProcessors);
                 deserializer.Read(out _expectedCPUSamplingRate);
+                // TODO: alig
             }
+#if SUPPORT_V1_V2
             else
             {
                 _processId = 0; // V1 && V2 tests expect 0 for process Id
                 pointerSize = 8; // V1 EventPipe only supports Linux which is x64 only.
                 numberOfProcessors = 1;
-
-                _startEventOfStream = deserializer.Current;      // Events immediately after the header.  
             }
+#endif
         }
 
-        int _fileFormatVersionNumber;
-        StreamLabel _startEventOfStream;
+#if SUPPORT_V1_V2
         StreamLabel _endOfEventStream;
-
-        Dictionary<StreamLabel, EventPipeEventMetaData> _eventMetadataDictionary = new Dictionary<StreamLabel, EventPipeEventMetaData>();
+#endif
+        int _fileFormatVersionNumber;
+        Dictionary<int, EventPipeEventMetaData> _eventMetadataDictionary = new Dictionary<int, EventPipeEventMetaData>();
         Deserializer _deserializer;
         EventPipeTraceEventParser _eventParser; // TODO does this belong here?
         string _processName;
         internal int _processId;
         internal int _expectedCPUSamplingRate;
-
-        #endregion
+#endregion
     }
 
-    #region private classes 
+#region private classes
+
+    /// <summary>
+    /// An EVentPipeEventBlock represents a block of events.   It basicaly only has
+    /// one field, which is the size in bytes of the block.  But when its FromStream
+    /// is called, it will perform the callbacks for the events (thus deserializing
+    /// it performs dispatch).  
+    /// </summary>
+    internal class EventPipeEventBlock : IFastSerializable
+    {
+        public EventPipeEventBlock(EventPipeEventSource source) => _source = source;
+
+        unsafe public void FromStream(Deserializer deserializer)
+        {
+            // blockSizeInBytes INCLUDES any padding bytes to insure alignment.  
+            var blockSizeInBytes = deserializer.ReadInt();
+            _startEventData = deserializer.Current;
+            _endEventData = _startEventData.Add(blockSizeInBytes);
+
+            // Dispatch through all the events.  
+            PinnedStreamReader deserializerReader = (PinnedStreamReader)deserializer.Reader;
+
+            // Align to a 4 byte boundary
```

Comment:
I assume that the way this works is that you aligned the EventPipeEventBlock as a whole so there was no need to do alignment inside the block.    

You can simply delete the *ptr variable however (since it is not used except for asserts), and add an assert that _startEventData is 4 byte aligned above).